src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+import fs from 'fs';
+import commander from 'commander';
+import { PolykeyAgent, PeerInfo } from '../lib/Polykey';
+import {} from '../../tmp/hole-punch-test/publicserver';
+import { actionRunner, pkLogger, PKMessageType, determineNodePath } from '.';
+
+function makeAddPeerCommand() {
+  return new commander.Command('add')
+    .description('add a new peer to the store')
+    .option('-b, --base64 <base64>', 'decode the peer info from a base64 string')
+    .option('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-a, --connected-address <connectedAddress>', 'public address on which the node can be contacted')
+    .option('-r, --relay-key <relayKey>', 'path to the file which contains the public key of the relay peer')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const base64String = options?.base64?.replace('\r', '')?.replace('\n', '');
+
+        let successful: boolean;
+        if (base64String != undefined) {
+          // read in peer info string
+          const { publicKey, connectedAddress, relayPublicKey } = PeerInfo.parseB64(base64String);
+          successful = await client.addPeer(nodePath, publicKey, connectedAddress?.toString() ?? '', relayPublicKey);
+        } else {
+          const publicKeyPath = options.publicKey;
+          // read in publicKey if it exists
+          const publicKey = fs.readFileSync(publicKeyPath).toString();
+
+          const connectedAddress = options.connectedAddress;
+          const relayPublicKeyPath = options.relayKey;
+          const relayPublicKey = fs.readFileSync(relayPublicKeyPath).toString();
+
+          successful = await client.addPeer(nodePath, publicKey, connectedAddress, relayPublicKey);
+        }
+
+        if (successful) {
+          pkLogger('peer successfully added to peer store', PKMessageType.SUCCESS);
```

Comment:
Are these meant to be output messages or STDERR messages? Usually STDOUT is reserved for the "output' of the command. So the command here is to add a peer. The output should be some value representing the existence of a new peer. At the same time exit codes can then be used as well. The output shouldn't be arbitrary human readable messages, that should be going to STDERR.

This makes it easier to use the commands in a non-interactive way.